### PR TITLE
feat: Testing Documentation

### PR DESCRIPTION
**Pull request by giovannivignone**
ref: #235 

- Adding testing documentation to the ReadMe.md
- Adding a cargo testall command to run all tests in all workspaces/packages
- Adding a .env.example to make it easy for devs to load envs into their tests
- Adding dotenv to dev dependencies

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,3 @@
+CONSENSUS_RPC=https://www.lightclientdata.org
+MAINNET_RPC=https://eth-mainnet.g.alchemy.com/v2/XXXXX
```
Comment (ncitron) [inline .env.example:2]: Should we differentiate between rpc types?

Wouldn't it make more sense to have:
MAINNET_CONSENSUS_RPC
MAINNET_EXECUTION_RPC
GOERLI_CONSENSUS_RPC
GOERLI_EXECUTION_RPC

**Review by ncitron — COMMENTED** [review]
Mostly looks good.

Can you also make sure that if a test uses the env vars, if the env var is not set, just skip it rather than failing?

PRs made from forks cannot access a repos secrets, so the lack of these vars will cause all those external PRs to fail CI, which isn't great. Short term I think skipping them is fine (it will run anyway when it is merged to master though).

File: .env.example (hunk 1 above)
Comment (giovannivignone) [reply .env.example]: Done @ncitron

**Review by ncitron — APPROVED** [review]
LGTM!

**Pull request merged**